tests/test_git/common.py

```diff
@@ -0,0 +1,60 @@
+import sublime
+import subprocess
+import os
+import shutil
+import tempfile
+from unittesting import DeferrableTestCase
+
+
+# copy from https://github.com/randy3k/ProjectManager/blob/f8667697a77a9e7020891e3227de25c2ebb990b3/pm.py#L72
+def subl(args=[]):
+    executable_path = sublime.executable_path()
+    if sublime.platform() == 'osx':
+        app_path = executable_path[:executable_path.rfind(".app/") + 5]
+        executable_path = app_path + "Contents/SharedSupport/bin/subl"
+    subprocess.Popen([executable_path] + list(args))
+    if sublime.platform() == "windows":
+        def fix_focus():
+            window = sublime.active_window()
+            view = window.active_view()
+            window.run_command('focus_neighboring_group')
+            window.focus_view(view)
+        sublime.set_timeout(fix_focus, 300)
+
+
+class GSCreateRepoTestCase(DeferrableTestCase):
+
+    _temp_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup a repo for testing
+        """
+        cls._temp_dir = tempfile.mkdtemp()
+        subl(["-n", cls._temp_dir])
+
+        def condition():
+            for d in sublime.active_window().folders():
+                # on Windows, `cls._temp_dir` is lowered cased,
+                # `os.path.normcase` is needed for comparison.
+                if cls._temp_dir == os.path.normcase(d):
+                    return True
+
+        yield condition
+
+        cls.window = sublime.active_window()
+
+        print(subprocess.check_output(["git", "--version"]).decode("ascii").strip())
+        subprocess.check_call(["git", "init"], cwd=cls._temp_dir)
+
+    @classmethod
+    def tearDownClass(cls):
+        # need at least one window in order to keep sublime running
+        if len(sublime.windows()) > 1:
+            cls.window.run_command("close")
+            yield 1000
```

Comment:
Again I find this hard to read. Why are we yielding an integer here but a callable in setUpClass? Does this integer cause some kind of sleep ?